### PR TITLE
Add missing collateral fields to `ProtocolParamUpdate`

### LINKED ISSUE #1296: ProtocolParamUpdate missing vasil collateral fields
An oversight, both the CSL (`src/Internal/Serialization/Types.purs`) and our Cardano (`src/Internal/Cardano/Types/Transaction.purs`) `ProtocolParamUpdate` types are missing `collateral_percentage` and `max_collateral_inputs` and the associated functions to work with them.

### PR DESCRIPTION
**Pull request by pb99u069**
Closes #1296.

### Pre-review checklist

- [ ] All code has been formatted using our config (`make format`)
- [ ] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: test/Fixtures.purs [hunk 1]
```diff
@@ -321,6 +321,8 @@ proposedProtocolParameterUpdates1 = ProposedProtocolParameterUpdates $
         , maxBlockExUnits: Just
             { mem: BigInt.fromInt 1, steps: BigInt.fromInt 1 }
         , maxValueSize: Just $ UInt.fromInt 1
+        , collateralPercentage: Nothing -- Just $ UInt.fromInt 140
+        , maxCollateralInputs: Nothing -- Just $ UInt.fromInt 10
```
Comment (pb99u069) [inline test/Fixtures.purs:325]: @klntsky I can't figure out why the unit test `Deserialization is inverse to serialization #4` fails when I set the two new fields to a `Just` value. The error msg is `Cannot deserialize bytes`. meanwhile, if the fields are `Nothing`, the test passes

Comment (klntsky) [reply test/Fixtures.purs:325]: @pb99u069 use `spy` function to debug this (you can detect where decoding fails by inserting it in a pure expression)

Comment (jy14898) [reply test/Fixtures.purs:325]: If you print the error message CSL returns in `fromBytes`, it seems confused and thinks the value we pass as a collateral_percentage is a key in some map. So either we're doing something wrong on the to_bytes/CSL building side, or CSL has a bug

Comment (pb99u069) [reply test/Fixtures.purs:325]: The functions `set_collateral_percentage` and `set_max_collateral_inputs` [here](https://github.com/Emurgo/cardano-serialization-lib/blob/fa2a9c94bcd1e3a7353852915a0f2e666ec0027f/rust/src/lib.rs#L2690) differ from all the other setters, as they don't use `clone()` at the end. Maybe that's our problem?

Comment (jy14898) [reply test/Fixtures.purs:325]: I saw that too but there are other cases that aren't cloned, `max_tx_size` for example

Comment (klntsky) [reply test/Fixtures.purs:325]: @jy14898 
> it seems confused and thinks the value we pass as a collateral_percentage is a key in some map

Could you please post the message for me to find where is it thrown from?

Comment (jy14898) [reply test/Fixtures.purs:325]: Definitely a bug with CSL:

```
> ppu = lib.ProtocolParamUpdate.new()
ProtocolParamUpdate { ptr: 1668248 }
> ppu.set_collateral_percentage(42)
undefined
> ppu.collateral_percentage()
42
> lib.ProtocolParamUpdate.from_bytes(ppu.to_bytes()).collateral_percentage()
undefined
```

Comment (jy14898) [reply test/Fixtures.purs:325]: @klntsky I used the following to always print the error, as we hush it in purescript. But the error is misleading due to nested decoding, the crux of the problem is as I posted above

```diff
--- a/src/Internal/Deserialization/FromBytes.js
+++ b/src/Internal/Deserialization/FromBytes.js
@@ -11,6 +11,7 @@ const fromBytes = name => helper => bytes => {
   try {
     return helper.valid(lib[name].from_bytes(bytes));
   } catch (e) {
+    console.log(name + ".from_bytes() raised " + e);
     return helper.error(name + ".from_bytes() raised " + e);
   }
 };
```

Comment (klntsky) [reply test/Fixtures.purs:325]: Made an issue
https://github.com/Emurgo/cardano-serialization-lib/issues/564

Comment (pb99u069) [reply test/Fixtures.purs:325]: https://github.com/Emurgo/cardano-serialization-lib/pull/565

Comment (klntsky) [reply test/Fixtures.purs:325]: Yep, but the release hasn't been cut yet (I hope it will be soon)

**Review by jy14898 — APPROVED** [review]
LGTM. Not sure if we need a changelog entry for this, given nobody filed a bug for it

**Review by jy14898 — CHANGES_REQUESTED** [review]
Actually, we need to update the template too

File: package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
     "@emurgo/cardano-serialization-lib-browser": "11.2.1",
     "@emurgo/cardano-serialization-lib-nodejs": "11.2.1",
     "base64-js": "^1.5.1",
+    "@noble/secp256k1": "^1.7.0",
```
Comment (pb99u069) [inline package.json:37]: It's strange that this dependency was missing. I guess we need it?

**Comment by pb99u069** [comment]
when I run `npm i -packages-lock-only` in the template I get the warning that there are 9 vulnerabilites. Which is strange as we have the same dependencies as in the main package.json. Any idea why this could be @jy14898?

**Comment by jy14898** [comment]
@pb99u069 Maybe we dealt with these package-lock issues for CTL but not the template, we'd need to diff the locks to find out where they don't agree

**Review by klntsky — APPROVED** [review]
Thanks for debugging this tricky one!

**Review by jy14898 — APPROVED** [review]

**Pull request merged**